nntrainer/layers/layer_node.cpp

```diff
@@ -322,18 +344,33 @@ nntrainer::Layer *LayerNode::getLayer() {
 
 void LayerNode::updateInputLayers(const std::string &from,
                                   const std::string &to) {
-  for (unsigned int idx = 0; idx < input_layers.size(); ++idx) {
-    if (istrequal(input_layers[idx], from)) {
-      input_layers[idx] = to;
+  auto &input_layers =
+    std::get<std::vector<props::InputLayer>>(*layer_node_props);
+  for (auto &input_layer : input_layers) {
+    if (istrequal(input_layer, from)) {
+      input_layer.set(to);
     }
   }
 }
 
 void LayerNode::updateInputLayers(const unsigned int idx,
                                   const std::string &to) {
-  if (idx >= input_layers.size())
-    throw std::out_of_range("Out of range for input_layers");
-  input_layers[idx] = to;
+  auto &input_layers =
+    std::get<std::vector<props::InputLayer>>(*layer_node_props);
+
+  input_layers.at(idx).set(to);
+}
+
+void LayerNode::addInputLayers(const std::string &in_layer) {
+  auto &input_layers =
+    std::get<std::vector<props::InputLayer>>(*layer_node_props);
+  input_layers.emplace_back(in_layer);
+}
+
+void LayerNode::setInputLayers(const std::vector<std::string> &layers) {
+  auto &input_layers =
```

Comment:
consider adding an assert that numInputDimensions must always be >= number of input connections.

Reply:
https://github.com/nnstreamer/nntrainer/pull/1542#discussion_r704021846